engine/select_members.py

```diff
@@ -10,146 +10,161 @@
-                ]
-                passed, new_vars = test_selection(
-                    stats_file_name,
-                    random_tolerance_file_name,
-                    validation_members,
-                    member_type,
-                    f,
-                )
-
-                validation_members_np = np.array(validation_members)
-                indices = [i for i, value in enumerate(passed) if value == 0]
-                failed = validation_members_np[indices]
-                # Increase weights for members which failed
-                weights[failed - 1] += 1 / total_member_num
-                # weights needs to sum up to 1 for np.random.choice
-                weights = weights / sum(weights)
-
-                variables.extend(new_vars)
-
-                if (mem_num == max_member_num) and (f == max_factor):
-                    duplicates = dict(Counter(variables).items())
-                    sorted_duplicates = dict(
-                        sorted(duplicates.items(), key=lambda x: x[1], reverse=True)
-                    )
-                # The following is to save computing time
-                elif iteration < 32:
-                    max_passed = max(max_passed, sum(passed))
-                    # The more combs were tested
-                    # the higher should the success rate be to continue
-                    tested_stats = len(validation_members)
-                    if max_passed < (iteration + 1) * 0.03 * tested_stats:
-                        break
-
-                if sum(passed) == len(validation_members):
-                    return random_members, f
-        # If factore needs to be increased, test only with max_members
-        min_member_num = max_member_num
-
-    max_count = max(sorted_duplicates.values())
-    most_common_vars = [
-        item for item, count in sorted_duplicates.items() if count == max_count
-    ]
     logger.error(
-        "ERROR: Could not find %s random members, which pass for all stat files. "
-        + "The most sensitive variable(s) is/are %s, which failed for %s out of %s "
-        + "random selections. Consider removing this/those variable(s) from the "
+        "ERROR: Could not find %s members, which pass for all stat files. "
+        + "The most sensitive variable(s) is/are %s, which failed with the factor %s"
+        + ". Consider removing this/those variable(s) from the "
         + "experiment and run again.",
-        max_member_num,
+        max_member_count,
         most_common_vars,
-        max_count,
-        iteration + 1,
+        max_factor,
     )
     sys.exit(1)
 
 
-# Tests how may stats files pass the tolerance test for the selected members
-# Returns the number of passed stats files and the variables which failed
 def test_selection(
-    stats_file_name, tolerance_file_name, total_member_num, member_type, factor
+    stats_file_name, tolerance_file_name, total_member_count, member_type, factor
```

Comment:
Actually I don't like that we use `total_member_count` here because it can also be only the members that should be tested, e.g., the `validation_members` as given as an input in `find_members_and_factor_validating_for_all_stats_files`. Again, this can actually be a number or a list. Sorry that goes back to me...

Reply:
Yes, I think in the current state we actually only use this parameter as a list. Would it be ok, to rename `total_member_count` into `member_ids`, and remove the possibilty to pass an `int` here?

Reply:
There is still the option `test_tolerance` in the function `select_members`, which gives an integer as input. But maybe we can get rid of this option. If we don't want to get rid of this option, we would have to add `member_ids` as optional input and use this for the `test_tolerance` option.

Otherwise I'm find with removing the possibility to pass `int` to `test_selection`.